joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/MethodRefLinker.scala

```diff
@@ -56,6 +54,12 @@ object MethodRefLinker {
   def nodesWithFullName(cpg: Cpg, x: String): mutable.Seq[NodeRef[_ <: NodeDb]] =
     cpg.graph.indexManager.lookup(PropertyNames.FULL_NAME, x).asScala
 
+  /** Types from C/C++ can be annotated with * to indicate being a reference. As our CPG schema currently lacks a
+    * separate field for that information the * is part of the type full name and needs to be removed when linking.
+    */
+  private def dereferenceTypeFullName(fullName: String): String =
```

Comment:
this would feel better to me if we moved it into an `if language == newc` or a c-specific subclass

Reply:
Good point.